### ISSUE TITLE
apps/glsl is failing

### ISSUE THREAD

**Issue by steven-johnson** [halide/Halide#3371]
$ make distrib && cd apps/glsl && make run
Internal error at CodeGen_GPU_Host.cpp:54 triggered by user code at ../../distrib/tools/GenGen.cpp:4:
Condition failed: is_zero(op->min)

**Comment by steven-johnson** [comment]
(Note that this app has no `test` target, as it is supposed to, thus `make test_apps` is skipping it)

**Comment by shoaibkamil** [comment]
@abadams After bisection:
```
b3499c7bb85cd6aa916efbc7b67875d8d3f63f12 is the first bad commit
commit b3499c7bb85cd6aa916efbc7b67875d8d3f63f12
Author: Andrew Adams <andrew.b.adams@gmail.com>
Date:   Fri Oct 5 10:57:19 2018 -0700

    Delay canonicalize_gpu_vars to try to fix #3221
```

I'll take a look this week to see if I can fix this and make this test run as part of `make test_apps`.

**Comment by alexreinking** [comment]
This issue breaks the cmake build, which does have a target for building it.
